os/20201103-full-disk-raid.md

```diff
@@ -0,0 +1,208 @@
+# Full disk RAID support
+
+## Theory of operation
+
+Users write an FCC section specifying the disks that should participate in redundant boot.  FCCT desugars that section into an Ignition config which completely recreates the partition structure on each specified disk.  The config creates partitions, RAID volumes, and filesystems for the root and boot partitions; partitions and independent filesystems for each ESP replica; and partitions for each BIOS-BOOT replica.
+
+Dracut glue detects these config stanzas, saves the corresponding on-disk data to RAM before the Ignition disks phase, and restores it after disks phase completes.  The glue also replicates the boot sector when replicating BIOS-BOOT.
+
+Users should only create replicated volumes using the official FCC sugar.  However, each piece of the replication logic (root partition, boot partition, ESP, and BIOS-BOOT + MBR) functions independently, and an advanced user can create nonstandard partition configurations (for example, RAID-1 boot with RAID-5 root) by manually configuring partitions in their FCC.
+
+## FCC sugar
+
+```yaml
+variant: fcos
+version: 1.3.0
+storage:
+  boot_raid:
+    devices:
+      # Two or more devices are allowed
+      - /dev/vda
+      - /dev/vdb
+    # Optional, defaults to x86_64
+    layout: x86_64
+    # Optional, defaults to 0 (entire disk)
+    # Do we want this?
+    root_size_mib: 8192
+    # Optional, defaults to xfs
+    # Do we want this?
+    root_format: xfs
```

Comment:
Is the thinking with these that there's no easy way for users to specify it themselves via `storage.disks.partitions[]`? (Because IIUC, they'd have to know the label names the sugar generates.)